ui/tests/integration/components/tools/wrap-test.js

```diff
@@ -104,6 +109,67 @@ module('Integration | Component | tools/wrap', function (hooks) {
     await click(TS.submit);
   });
 
+  test('it toggles between views and preserves input data', async function (assert) {
+    assert.expect(6);
+    await this.renderComponent();
+    await codemirror().setValue(this.wrapData);
+    assert.dom('[data-test-component="json-editor-title"]').hasText('Data to wrap (json-formatted)');
+    await click('[data-test-toggle-input="json"]');
+    assert.dom('[data-test-component="json-editor-title"]').doesNotExist();
+    assert.dom('[data-test-kv-key="0"]').hasValue('foo');
+    assert.dom('[data-test-kv-value="0"]').hasValue('bar');
+    await click('[data-test-toggle-input="json"]');
+    assert.dom('[data-test-component="json-editor-title"]').exists();
+    assert.strictEqual(
+      codemirror().getValue(' '),
+      `{   \"foo": \"bar" }`, // eslint-disable-line no-useless-escape
+      'json editor has original data'
+    );
+  });
+
+  test('it submits from kv view', async function (assert) {
+    assert.expect(6);
+
+    const multilineData = `this is a multi-line secret
+      that contains
+      some seriously important config`;
+    const flashSpy = sinon.spy(this.owner.lookup('service:flash-messages'), 'success');
+    const updatedWrapData = JSON.stringify({
```

Comment:
Is there a reason to stringify?  We can just compare the two objects below, right? (Instead of need to call `JSON.parse()`